cmd/scaffold.go

```diff
@@ -530,12 +539,58 @@ func (fnb *FlowNodeBuilder) initMetrics() {
 	}
 }
 
+func (fnb *FlowNodeBuilder) createUploader(client *gcemd.Client, opts ...option.ClientOption) (profiler.Uploader, error) {
+	projectID, err := client.ProjectID()
+	if err != nil {
+		fnb.Logger.Warn().Err(err).Msg("could not get CGE project ID")
+		projectID = "unknown"
+	}
+
+	instance, err := client.InstanceID()
+	if err != nil {
+		fnb.Logger.Warn().Err(err).Msg("could not get CGE instance ID")
+		instance = "unknown"
+	}
+
+	chainID := fnb.RootChainID.String()
+	if chainID == "" {
+		chainID = "unknown"
+	}
+
+	params := profiler.Params{
+		ProjectID: projectID,
+		ChainID:   chainID,
+		Role:      fnb.NodeConfig.NodeRole,
+		Version:   build.Semver(),
+		Commit:    build.Commit(),
+		Instance:  instance,
+	}
+
+	fnb.Logger.Info().Msgf("creating pprof profile uploader with params: %+v", params)
+
+	return profiler.NewUploader(fnb.Logger, params, opts...)
+}
+
 func (fnb *FlowNodeBuilder) initProfiler() {
```

Comment:
not specific to your PR, but ignore errors (only logging warning) seems kind of terrible.  we should propagate the error whenever possible ...